### PR TITLE
samples: net: Check the return value of nats_publish

### LINKED ISSUE #6258: [Coverity CID: 182894] Error handling issues in /samples/net/nats/src/main.c
Static code scan issues seen in File: /samples/net/nats/src/main.c
 Category: Error handling issues
 Function: write_led
 Component: Samples
 CID: 182894
 Please fix or provide comments to square it off in coverity in the link: https://scan9.coverity.com/reports.htm#v32951/p12996

### PR DESCRIPTION
**Pull request by satyacube**
Check and handle the return value of nats_publish() in main.c. Also, apply
the changes suggested by the tools uncrustify and checkpatch.

Signed-off-by: Satya Bhattacharya <satyacube@gmail.com>

Fixes #6258.

### CODE REVIEW

**Comment by pfalcon** [comment]
This appears to duplicate work done for https://github.com/zephyrproject-rtos/zephyr/pull/8175

**Review by jukkar — CHANGES_REQUESTED** [review]
Your changes look ok but you should only have one logical change in one commit. So please separate the return value check and style changes into separate commits (they can still be part of this PR).

Also have signed-off-by line as last line in the commit body.

**Review by lpereira — CHANGES_REQUESTED** [review]
Please rebase your changes on top of HEAD.  Your pull request contains a lot of unrelated changes.

File: samples/net/nats/src/main.c [hunk 1]
```diff
@@ -16,7 +16,7 @@
 /* LED */
 #ifndef LED0_GPIO_CONTROLLER
 #ifdef LED0_GPIO_PORT
-#define LED0_GPIO_CONTROLLER 	LED0_GPIO_PORT
+#define LED0_GPIO_CONTROLLER    LED0_GPIO_PORT
```
Comment (jukkar) [inline samples/net/nats/src/main.c]: Unrelated change, please remove this part from this commit.

**Review by jukkar — CHANGES_REQUESTED** [review]

File: samples/net/nats/src/main.c (hunk 1 above)
Comment (satyacube) [reply samples/net/nats/src/main.c]: I actually tried to undo all the style related changes. Unfortunately, I was successful in reverting all of them but one-git lint was unhappy with the formatting of the above line (baseline formatting was incorrect)
Is there any way around this? How can I drop all the changes which were made in my initial commit?  Thanks for your help.

File: samples/net/nats/src/main.c [hunk 2]
```diff
@@ -202,10 +203,13 @@ static void write_led(const struct nats *nats,
...
 
 	pubstate = state ? "on" : "off";
-	nats_publish(nats, "led0", 0, msg->reply_to, 0,
-		     pubstate, strlen(pubstate));
-
-	printk("*** Turning LED %s\n", pubstate);
+	ret = nats_publish(nats, "led0", 0, msg->reply_to, 0,
+			   pubstate, strlen(pubstate));
+	if (ret < 0) {
+		NET_DBG("Failed to publish: %d", ret);
```
Comment (pfalcon) [inline samples/net/nats/src/main.c]: Please don't use NET_DBG here - it's not a debugging message, it's real error. I'd recommend to just use printk(), like used below.

**Review by pfalcon — CHANGES_REQUESTED** [review]

File: samples/net/nats/src/main.c (hunk 1 above)
Comment (pfalcon) [reply samples/net/nats/src/main.c]: > How can I drop all the changes which were made in my initial commit?

Given that the "real" changes required for this PR are small, you can just drop this change, and re-do it from scratch. That may as well be the quickest way. Otherwise, the way to deal with these is to add a new commit reverting a particular change (e.g. line) and squash it into previous commit with `git rebase --interactive`. Hope this helps.

Comment (satyacube) [reply samples/net/nats/src/main.c]: Yes, it did work. Hopefully, now the changes are good for commit.

**Review by jukkar — APPROVED** [review]
LGTM

**Review by pfalcon — APPROVED** [review]
Thanks for cleanup!

**Comment by satyacube** [comment]
@lpereira Let me know if any more changes are required. If not, can you approve these changes?

**Review by lpereira — APPROVED** [review]
LGTM

**Review by MaureenHelm — APPROVED** [review]
